qa/L0_batch_input/batch_input_test.py

```diff
@@ -244,6 +244,93 @@ def test_max_element_count_as_shape(self):
         except InferenceServerException as ex:
             self.assertTrue(False, "unexpected error {}".format(ex))
 
+    def test_batch_item_shape_flatten(self):
+        # Use 4 set of inputs with shape
+        # [1, 4, 1], [1, 1, 2], [1, 1, 2], [1, 2, 2]
+        self.inputs = []
+        for value in [[1, 4, 1], [1, 1, 2], [1, 1, 2], [1, 2, 2]]:
+            self.inputs.append(
+                [tritonhttpclient.InferInput('RAGGED_INPUT', value, "FP32")])
+            self.inputs[-1][0].set_data_from_numpy(
+                np.full(value, value[0], np.float32))
+        self.client = tritonhttpclient.InferenceServerClient(
+            url="localhost:8000", concurrency=len(self.inputs))
+
+        model_name = "batch_item_flatten"
+
+        output_name = 'BATCH_OUTPUT'
+        outputs = [tritonhttpclient.InferRequestedOutput(output_name)]
+
+        async_requests = []
+        try:
+            for inputs in self.inputs:
+                # Asynchronous inference call.
+                async_requests.append(
+                    self.client.async_infer(model_name=model_name,
+                                            inputs=inputs,
+                                            outputs=outputs))
+
+            expected_value = np.asarray([[4, 1, 1, 2, 1, 2, 2, 2]], np.float32)
+            for idx in range(len(async_requests)):
+                # Get the result from the initiated asynchronous inference request.
+                # Note the call will block till the server responds.
+                result = async_requests[idx].get_result()
+
+                # Validate the results by comparing with precomputed values.
+                output_data = result.as_numpy(output_name)
+                self.assertTrue(
+                    np.array_equal(output_data, expected_value),
+                    "Expect response {} to have value {}, got {}".format(
+                        idx, expected_value, output_data))
+        except InferenceServerException as ex:
+            self.assertTrue(False, "unexpected error {}".format(ex))
+
+    def test_batch_item_shape(self):
+        # Use 3 set of inputs with shape [2, 1, 2], [1, 1, 2], [1, 2, 2]
+        inputs = []
+        for value in [[2, 1, 2], [1, 1, 2], [1, 2, 2]]:
+            inputs.append(
+                [tritonhttpclient.InferInput('RAGGED_INPUT', value, "FP32")])
+            inputs[-1][0].set_data_from_numpy(
+                np.full(value, value[0], np.float32))
```

Comment:
Same here. Does the value of the inputs matter?

Reply:
I'd change the np.full to be np.ones or hardcode the value being filled instead of making it value[0]

Reply:
What is the reason for this change? If it is to highlight the value is irrelevant, I think the better way is to clarify it with a comment, using hardcode value may give the opposite impression (i.e. a special value is required)

Reply:
Sure, you can use a comment to clarify the same.